lib/CStack.cpp

```diff
@@ -167,7 +167,7 @@ std::string CStack::nodeName() const
 	oss << owner.toString();
 	oss << " battle stack [" << ID << "]: " << getCount() << " of ";
 	if(typeID.hasValue())
-		oss << typeID.toEntity(VLC)->getNamePluralTextID();
+		oss << typeID.toCreature()->getNamePluralTranslated();
```

Comment:
Access to translated name is relatively slow, and some of these methods are used A LOT. Json key should be faster and still be readable (e.g. `core:archer`)
```suggestion
		oss << typeID.toEntity(VLC).getJsonKey();
```

Reply:
Ok, it's just... those were super long. I will check more on that

Reply:
Done as suggested. Also to one more BattleAI logging place.